### ISSUE TITLE
About the implementation of the randomized convolutional layer

### ISSUE THREAD

**Issue by gulai22** [pokaxpoka/netrand#5]
I tried to use the randomized convolution layer on my model to improve the robustness, however, I don't understand why you use a `mask_vbox` to calculate `img` in your [implementation](https://github.com/pokaxpoka/netrand/blob/master/sources/policies.py). Why not just input the original `images` to the randomized convolution layer? Can you give me some advice?

```mask_vbox = tf.Variable(tf.zeros_like(images, dtype=bool), trainable=False)
mask_shape = tf.shape(images)
rh = .2 # hard-coded velocity box size
mh = tf.cast(tf.cast(mask_shape[1], dtype=tf.float32)*rh, dtype=tf.int32)
mw = mh*2
mask_vbox = mask_vbox[:,:mh,:mw].assign(tf.ones([mask_shape[0], mh, mw, mask_shape[3]], dtype=bool))

img  = tf.where(mask_vbox, x=tf.zeros_like(images), y=images)
rand_img = tf.layers.conv2d(img, randcnn_depth, 3, padding='same', kernel_initializer=tf.initializers.glorot_normal(), trainable=False, name='randcnn')
out = tf.where(mask_vbox, x=images, y=rand_img, name='randout')
```

**Comment by kibok90** [comment]
coinrun puts two velocity boxes on the top-left corner; you may check Appendix A in the coinrun paper or Appendix C in our paper.

These boxes are a kind of "visualized dynamics information," not a real visual information. So we do not randomize the boxes by masking them.
